Code/Framework/AzCore/AzCore/std/allocator_static.h

```diff
@@ -126,21 +107,13 @@ namespace AZStd
     template<AZStd::size_t Size, AZStd::size_t Alignment>
     AZ_FORCE_INLINE bool operator==(const static_buffer_allocator<Size, Alignment>& a, const static_buffer_allocator<Size, Alignment>& b)
     {
-        if (&a == &b)
-        {
-            return true;
-        }
-        return false;
+        return static_cast<bool>(&a == &b);
     }
 
     template<AZStd::size_t Size, AZStd::size_t Alignment>
     AZ_FORCE_INLINE bool operator!=(const static_buffer_allocator<Size, Alignment>& a, const static_buffer_allocator<Size, Alignment>& b)
     {
-        if (&a != &b)
-        {
-            return true;
-        }
-        return false;
+        return static_cast<bool>(&a != &b);
```

Comment:
repeat: Why the static_cast to bool?

Reply:
I just applied clang-tidy's [simplify-boolean-expr](https://clang.llvm.org/extra/clang-tidy/checks/readability/simplify-boolean-expr.html) fix-it, I'm not sure why it determined it should get a `static_cast`. It looks like that check is only supposed to add the `static_cast` if the comparison involves types that have an `explicit operator bool()`, which these do not. I'll take the cast out.

Reply:
Updated to remove the static cast.